### PR TITLE
feat(config): Validate parent of fileMatch field

### LINKED ISSUE #7490: Raise Config Error if `fileMatch` is configured outside manager block
**What would you like Renovate to be able to do?**

Detect a config error if `fileMatch` is configured anywhere except within a manager block. e.g. if configured at the top level or within a package rule.

### PR DESCRIPTION
**Pull request by zharinov**
Closes #7490

### CODE REVIEW

File: lib/config/validation.ts [hunk 1]
```diff
@@ -16,6 +17,15 @@ export interface ValidationResult {
   warnings: ValidationMessage[];
 }
 
+const managerList = getManagerList();
+
+function isManagerPath(parentPath: string): boolean {
+  return (
+    /^regexManagers\[[0-9]+]$/.test(parentPath) ||
+    managerList.includes(parentPath)
```
Comment (zharinov) [inline lib/config/validation.ts:25]: Did I miss something?

Comment (rarkins) [reply lib/config/validation.ts:25]: Now I'm worried that I forgot something else. Mostly I just don't want it configured at top level

**Review by viceice — DISMISSED** [review]
Looks good to me.

File: lib/config/validation.ts (hunk 1 above)
Comment (zharinov) [reply lib/config/validation.ts:25]: Then maybe error for top-level and warning for others?

Comment (rarkins) [reply lib/config/validation.ts:25]: I think it's ok as-is

**Review by rarkins — APPROVED** [review]
@viceice can you take one last look?

File: lib/config/validation.ts [hunk 2]
```diff
@@ -79,6 +89,19 @@ export async function validateConfig(
...
+      if (parentPath === undefined) {
+        errors.push({
+          depName: 'Config error',
+          message: `"fileMatch" may not be defined at the top level of a config and must instead be within a manager block`,
+        });
+      } else if (!isManagerPath(parentPath)) {
+        warnings.push({
+          depName: 'Config warning',
+          message: `"fileMatch" must be configured in a manager block and not here: ${parentPath}`,
+        });
```
Comment (viceice) [inline lib/config/validation.ts:102]: We should add an error here in future.

**Review by viceice — APPROVED** [review]
LGTM

**Pull request merged**